### PR TITLE
Initial working integration with companies house API.

### PR DESCRIPTION
**Pull request by dbgoaco**
[DP-478](https://noticingsystem.atlassian.net/jira/software/projects/DP/boards/22?selectedIssue=DP-478)

### CODE REVIEW

File: Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+using Flurl.Http;
+
+namespace CO.CDP.OrganisationApp.ThirdPartyApiClients.CompaniesHouse;
+
+public class CompaniesHouseApi(IConfiguration configuration, ILogger<CompaniesHouseApi> logger) : ICompaniesHouseApi
+{
+    public async Task<RegisteredAddress?> GetRegisteredAddress(
+        string companyNumber)
+    {
+        var companiesHouseUrl = GetComapniesHouseUrl();
```
Comment (andymantell) [inline Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs:11]: Typo - `Comapnies` should be `Companies`

**Review by andymantell — CHANGES_REQUESTED** [review]

File: Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs (hunk 1 above)
Comment (dbgoaco) [reply Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs]: Updated.

**Review by andymantell — APPROVED** [review]

**Comment by andymantell** [comment]
Tested locally - works as expected

File: Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
+using Flurl;
+using Flurl.Http;
+
+namespace CO.CDP.OrganisationApp.ThirdPartyApiClients.CompaniesHouse;
+
+public class CompaniesHouseApi(IConfiguration configuration, ILogger<CompaniesHouseApi> logger) : ICompaniesHouseApi
```
Comment (dharmverma) [inline Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs:6]: Can we have unit test for the class?

File: Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            logger.LogError(exc, "Failed during call to get company profile.");
+        }
+
+        return profile;
+    }
+
+    private string? GetCompaniesHouseUrl()
+    {
+        return configuration["CompaniesHouse:Url"];
```
Comment (dharmverma) [inline Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs:58]: Require handling when those appsettings are absent?

**Review by JBaigGoaco — APPROVED** [review]
Approved, looking good but there are few comments to resolve.

File: Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs (hunk 2 above)
Comment (dbgoaco) [reply Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs:6]: Will add to ticket [DP-614](https://noticingsystem.atlassian.net/jira/software/projects/DP/boards/22?selectedIssue=DP-614)

File: Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs (hunk 3 above)
Comment (dbgoaco) [reply Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouse.cs:58]: Will add to ticket [DP-614](https://noticingsystem.atlassian.net/jira/software/projects/DP/boards/22?selectedIssue=DP-614)

**Pull request merged**